src/NetEscapades.Configuration.Vault/VaultConfigurationExtensions.cs

```diff
@@ -236,10 +472,27 @@ public static class VaultConfigurationExtensions
             {
                 throw new ArgumentNullException(nameof(client));
             }
-            if (secretLocationPaths == null)
+            
+            if (vaultSecretMappings == null)
+            {
+                throw new ArgumentNullException(nameof(vaultSecretMappings));
+            }
+            
+            if (!vaultSecretMappings.Any())
+            {
+                throw new ArgumentException($"{nameof(vaultSecretMappings)} cannot be empty", nameof(vaultSecretMappings));
+            }
+            
+            var nonEmptyPrefixes = vaultSecretMappings
+                .Select(m => m.Prefix)
+                .Where(p => !string.IsNullOrWhiteSpace(p))
+                .ToArray();
+
+            if (nonEmptyPrefixes.Length != nonEmptyPrefixes.Distinct().Count())
```

Comment:
I wonder if we should _list_ the duplicate prefixes? 🤔 

```suggestion
            var duplicatePrefixes = vaultSecretMappings
                .Where(p => !string.IsNullOrWhiteSpace(p.Prefix))
                .GroupBy(x => x.Prefix)
                .Where(g => g.Count() > 1)
                .Select(g => g.Key)
                .ToList();
```

That way we could include them in the exception below?